ffi/src/scan.rs

```diff
@@ -330,17 +329,16 @@ struct ContextWrapper {
 #[no_mangle]
 pub unsafe extern "C" fn visit_scan_data(
     data: *mut EngineDataHandle,
-    selection_vector: KernelBoolSlice,
+    selection_vec: KernelBoolSlice,
     engine_context: NullableCvoid,
     callback: CScanCallback,
 ) {
-    let selection_vec = Vec::<bool>::from_slice(selection_vector);
+    let selection_vec = unsafe { selection_vec.as_ref() };
```

Comment:
nit: safety comment?

Reply:
though i suppose this is in unsafe extern anyways...

Reply:
Yeah, `visit_scan_data` safety comment eventually needs to be updated, but prob not as part of this PR